subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -401,7 +402,7 @@ STATIC int prepare_msg_for_send(struct lwm2m_message *msg)
 	/* set the default (small) buffer for sending blocks */
 	msg->cpkt.data = msg->msg_data;
 	msg->cpkt.offset = 0;
-	msg->cpkt.max_len = MAX_PACKET_SIZE;
+	msg->cpkt.max_len = sizeof(msg->msg_data);
 
 	payload = coap_packet_get_payload(&msg->body_encode_buffer, &len);
 	if (len <= CONFIG_LWM2M_COAP_MAX_MSG_SIZE) {
```

Comment:
That part would need an update too I believe, currently we only check if the packet payload fits in CONFIG_LWM2M_COAP_MAX_MSG_SIZE, but now it should take whole packet size into account (including header).  Need to compare against `msg->body_encode_buffer.offset` (I don't see any CoAP helper func to get full packet size)?

Reply:
I'll fix.